src/redux/features/batches/batchesAsyncThunks.ts

```diff
@@ -6,19 +6,38 @@ import { Batch, CreateBatchRequestPayload, NurseryTransfer } from 'src/types/Bat
 import { BatchData, BatchId, UpdateBatchRequestPayloadWithId } from 'src/services/NurseryBatchService';
 import AccessionService from 'src/services/AccessionService';
 import { NurseryBatchService } from 'src/services';
+import { RootState } from '../../rootReducer';
+import { SearchResponseAccession } from '../accessions/accessionsThunks';
 
 export type SavableBatch = (CreateBatchRequestPayload | UpdateBatchRequestPayloadWithId) & Batch;
 
 export const requestSaveBatch = createAsyncThunk(
   'batches/save',
-  async (request: { batch: SavableBatch; timezone?: string }, { rejectWithValue }) => {
-    const { batch, timezone } = request;
+  async (
+    request: { batch: SavableBatch; organizationId: number; timezone?: string },
+    { rejectWithValue, getState }
+  ) => {
+    const { batch, organizationId, timezone } = request;
 
     let response: (Response & BatchData) | (Response & BatchId) | undefined;
     let responseQuantities: Partial<Response> = { requestSucceeded: true, error: undefined };
 
     if (batch.id === -1) {
       if (batch.accessionId) {
+        // Extra validation to ensure an accession ID matches the given species
```

Comment:
This is the request flow validation

Reply:
Should we be ensuring a valid selection by way of populating correct entries in the form instead?
When a species is selected, the accession dropdown entries should only be relevant to the species.. and vice-versa.

Reply:
Yeah, that validation is already in place [here](https://github.com/terraware/terraware-web/pull/1958/files#diff-a4bbdce52d6392f56289439f849e8a49a344471c5da2b2fbbe950ca165159768R222) where is UI scrubbing / validating based on the species selection

Reply:
Got it, thanks 👍